assets/js/modal.js

```diff
@@ -31,12 +36,85 @@ function renderModal () {
   })
 }
 
+/**
+ * Trap focus in modal
+ * Only called on open modals
+ */
+function trapFocus (event) {
+  if (state.ignoreFocusChanges) return
+  var modal = qs(MODAL_SELECTOR)
+  if (modal.contains(event.target)) {
+    state.lastFocus = event.target
+  } else {
+    focusFirstDescendant(modal)
```

Comment:
I believe we could simplify this a bit by finding focusable elements on our own, something like this:

```js
const FOCUSABLE_SELECTOR = 'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])'

function firstFocusableDescendant(element) {
  return element.querySelector(FOCUSABLE_SELECTOR)
}

function lastFocusableDescendant(element) {
  const elements = element.querySelectorAll(FOCUSABLE_SELECTOR)
  return elements[elements.length - 1]
}
```

Reply:
This is better than iterating childNodes, but I guess the `attemptFocus` would still be useful when things are hidden via css, but still targetable (see the fix I made to the hidden liveview url input).

Reply:
I guess we can ignore that case given we control when things are hidden. I'll change it to your proposed implementation.

Reply:
We could get all focusable elements and find the first for which `attemptFocus` succeeds (and if that's the last element we would reverse the elements and do the same). But going the simpler route sounds fine as you said!